### PR TITLE
Metabuild: support non-code directories

### PR DESCRIPTION
**Pull request by gperciva**

### CODE REVIEW

**Comment by gperciva** [comment]
This allows us to have a directory which doesn't compile any program.  Primarily intended for test code, such as kivaloo's `tests/kvlds-dump`.

This is based on #169, and I'd like to get that merged before having a final copy of this PR.  But you can see that it produces a nice `tests/buildnothing/Makefile`.

**Comment by gperciva** [comment]
Update PR, candidate for merge.

File: tests/buildnothing/Makefile.BSD [hunk 1]
```diff
@@ -0,0 +1,4 @@
+test:
+	echo "Exciting Makefile, huh?  Probably run a script here?"
```
Comment (cperciva) [inline tests/buildnothing/Makefile.BSD]: This should be `echo "SUCCESS"`, I think?

Anything which is hooked into the tests should behave like a test...

Comment (gperciva) [reply tests/buildnothing/Makefile.BSD]: well, it's not being run by `make test` (which is now `tests/test_libcperciva.sh`, so I figured it didn't matter.  But ok.

**Comment by gperciva** [comment]
Updated PR

**Pull request merged**